### ISSUE TITLE
[Skia] ScrollViewer.VerticalScrollBarVisibility does not show when no mouse is available

### ISSUE THREAD

**Issue by jeromelaban** [unoplatform/uno#14474]
### Current behavior

```xml
  <ScrollViewer HorizontalScrollBarVisibility="Visible"
                VerticalScrollBarVisibility="Visible">
```

Does not show the scrollbar until the mouse is moved (this can happen on a touch-screen only app).

### Expected behavior

The scrollbars show regardless of mouse mouvement.

### Works on UWP/WinUI

None

**Comment by jeromelaban** [comment]
@HakanL if you do have access to the scrollviewer, can you try doing this:

```csharp
VisualStateManager.GoToState(yourScrollViewer, "TouchIndicator", true);
```

This should show the scrollbar as a workaround.

**Comment by HakanL** [comment]
Thanks @jeromelaban that worked! I had to put it in the Loaded event (and not constructor like I tried first), but that may be expected, just wanted to document that if somebody else ran into this. Is there perhaps a way to do the same thing to get the larger scrollbar that shows up when you hover over the actual scroll bar?

**Comment by ramezgerges** [comment]
This isn't a bug, but is actually how WinUI behaves too. `Visiblity=Visible` doesn't permanently show the scrollbars, but just makes them showable on hovering.

However, a workaround would be to modify the template of the ScrollBar to basically always be in the VisualState we want (and subsequently modify the ScrollViewer template to use that ScrollBar template). After tinkering around for a bit, I realized that jumping to the exanded state of the ScrollBar is not enough. In reality, one has to hover over the ScrollViewer before then hovering over the ScrollBar. So there are actually 2 states in play: "MouseIndicator" (or "TouchIndicator", both are quite similar) and "Expanded". Playing around with the template worked but was quite messy and too fragile in case we (or Microsoft) updated the templates later. I've finally arrived at a simpler solution:

```csharp
#if HAS_UNO
        var vsb = (Control)sv.FindName("VerticalScrollBar");

        var expandScrollBar = (object _, object _) => {
            VisualStateManager.GoToState(vsb, "MouseIndicator", true);
            VisualStateManager.GoToState(vsb, "Expanded", true);
        };

        vsb.PointerExited += new Microsoft.UI.Xaml.Input.PointerEventHandler(expandScrollBar);
        vsb.PointerCaptureLost += new Microsoft.UI.Xaml.Input.PointerEventHandler(expandScrollBar);

        sv.PointerEntered += new Microsoft.UI.Xaml.Input.PointerEventHandler(expandScrollBar);
        sv.PointerMoved += new Microsoft.UI.Xaml.Input.PointerEventHandler(expandScrollBar);

        Loaded += new RoutedEventHandler(expandScrollBar);
#endif
```

Note however that this only works in Uno. On WinUI, `VerticalScrollBar` (and `HorizontalScrollBar`) are not exposed as UIElements. Instead, the visual tree directly contains the root `Grid` inside the ScrollBar template.

**Comment by HakanL** [comment]
@ramezgerges Is it correct to flip between the two states on all PointerMoved events? Or they are not sequential (MouseIndicator, Expanded)?

**Comment by ramezgerges** [comment]
This is all a hack, so I'm not sure about "correct", but it is a sound workaround, yes. Settings a VisualState just applies its `Setter`s (+ transitions, animations, etc), so all we're doing here is applying the setters from both visual states. This is what happens in the original scenario as well: you first hover over the ScrollViewer, which sets the visual state of the ScrollBar to `XXIndicator`, then you hover over the ScrollBar, which then switches to `Expanded`. My initial workaround was copying both VisualStates into a new bigger one and using that in a new template (which is perhaps how Microsoft would want you to do it), but this way, the workaround will keep working if the default templates are updated.

**Comment by HakanL** [comment]
Makes sense. Yes, it works great, now my only issue is that the visual focus rectangle is on top of the scrollbar, but I'm trying to figure out the styling to solve that. Thanks for the hack though, for a touch screen this is key.

**Comment by HakanL** [comment]
@ramezgerges   Now with Uno 6, do you think we can revisit this? It would be great to have a way to always display the scrollbars (without a mouse), without the `GoToState` hack.
